File: src/main/java/com/example/probation/core/entity/Role.java
package com.example.probation.core.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.springframework.security.core.GrantedAuthority;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@EqualsAndHashCode(of = "roleName", callSuper = false)
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "roles")
@Data
public class Role extends AbstractEntity implements GrantedAuthority {
    @Column(name = "role_name")
    private String roleName;

    @Override
    public String getAuthority() {
        return getRoleName();
    }
}

File: src/main/kotlin/com/example/probation/core/entity/Role.kt
package com.example.probation.core.entity

import org.springframework.security.core.GrantedAuthority
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "roles")
class Role(
    var roleName: String? = null,
) : AbstractEntity(), GrantedAuthority {
    constructor(roleName: String? = null, id: Long? = null) : this() {
        this.roleName = roleName
        this.id = id
    }
    override fun getAuthority(): String? = roleName
}
